taichi/struct/snode_tree.cpp

```diff
@@ -2,6 +2,18 @@
 
 namespace taichi {
 namespace lang {
+namespace {
+
+void get_snodes_to_root_id_impl(const SNode &node,
+                                const int root_id,
+                                std::unordered_map<int, int> *map) {
```

Comment:
Is there a reason we use the `*`-way instead of `&`-way to pass the map?

Reply:
It's mostly Google C++ Coding Style -- it suggests that mutating params being specified in the end and passed in as pointers